src/__experimental__/components/search/search.style.js

```diff
@@ -6,106 +6,104 @@ import { baseTheme } from '../../../style/themes';
 
 const StyledSearch = styled.div`
-  ${({ theme }) => `border-bottom: 2px solid ${theme.search.passive};`}
-  ${({ isFocused, searchHasValue }) => css`
-    ${!isFocused && !searchHasValue && css`
+  ${({
+    isFocused,
+    searchWidth,
+    searchIsActive,
+    searchHasValue,
+    theme
+  }) => {
+    return css`
+      width: ${searchWidth ? `${searchWidth}px` : '100%'};
+      padding-bottom: 2px;
+      background-color: transparent;
+      border-bottom: 2px solid ${theme.search.passive};
+      display: inline-flex;
+      font-size: 14px;
+      font-weight: 700;
+      margin-bottom: 0px;
+      :hover {
+        border-bottom-color: ${theme.search.active};
+        cursor: pointer;
+      }
+      ${!isFocused && !searchHasValue && css`
+        ${StyledInputPresentation} {
+          border: 1px solid transparent;
+          color: rgba(0, 0, 0, 0.65);
+        }
+      `}
+      ${(isFocused || searchHasValue) && css`
+        border-color: transparent;
+        transition: background 0.2s ease;
+        color: rgba(0, 0, 0, 0.9);
+        :hover {
+          border-color: transparent;
+        }
+      `}
+      ${isFocused && !searchIsActive && css`
+        color: rgba(0, 0, 0, 0.9);
+      `}
+      
       ${StyledInputPresentation} {
-        border: 1px solid transparent;
-        color: rgba(0, 0, 0, 0.65);
+        flex: 1;
+        background-color: transparent;
+        font-size: 14px;
+        font-weight: 700;
+        padding-bottom: 2px;
+        padding-top: 1px; 
+        cursor: pointer;
       }
-    `}
-    ${(isFocused || searchHasValue) && css`
-      border-bottom: 2px solid transparent;
-      transition: border 0.2s ease, background 0.2s ease;
-      color: rgba(0, 0, 0, 0.9);
-      :hover {
-        border-bottom: none;
+      ${StyledFormField} {
+        flex: 1;
+        z-index: 10;
       }
-    `}
-  `}
-
-  ${({ isFocused, searchIsActive }) => css`
-    ${isFocused && !searchIsActive && css`
-      color: rgba(0, 0, 0, 0.9);
-    `}
-  `}
-
-  display: inline-flex;
-  font-size: 14px;
-  font-weight: 700;
-  margin-bottom: 0px;
-  :hover {
-    ${({ theme }) => css` 
-    border-bottom-color: ${theme.search.active};`}
-    cursor: pointer;
-  }
-
-  ${StyledInputPresentation} {
-    width: ${
-  ({
-    hasSearchButton, isFocused, searchIsActive, searchHasValue
-  }) => (
-    hasSearchButton && (isFocused || searchIsActive || searchHasValue) ? '335px;' : '375px;'
-  )};
-
-    font-size: 14px;
-    font-weight: 700;
-    padding-bottom: 2px;
-    padding-top: 1px; 
-    cursor: pointer;
-  }
-
-  ${StyledFormField} { 
-    z-index: 10;
-  }
-
-  ${StyledButton} { 
-    background-color: #255BC7;
-    cursor: pointer;
-    color: white;
-  }
-
-  ${StyledIcon} {
-    color: #668592;
-    width: 20px;
-    height: 20px;
-    cursor: pointer;
-    :hover {
-      color: rgba(0, 0, 0, 0.65);
-    };
-  }
+      ${StyledButton} { 
+        background-color: #255BC7;
```

Comment:
this should be theme color instead of hardcoded hex value, i think the  theme.colors.search.button  is what you looking for.

Reply:
In fact please can you make sure that any hardcoded value is present in theme and use that instead?

Reply:
I can indeed.

Reply:
Done.